src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,496 @@
+            import com.snowflake.snowpark_java.types.UsernamePassword;
+
+
+            class OutputRow {{
+                {self.create_output_row_class()}
+            }}
+
+            public class DataLoader{{
+            private final Connection conn;
+
+            private static Connection createConnection() {{
+                try {{
+                    {self.generate_create_connection()}
+                }} catch (Exception e) {{
+                    throw new RuntimeException("Failed to create JDBC connection: " + e.getMessage());
+                }}
+            }}
+
+            public static Class<?> getOutputClass() {{
+                return OutputRow.class;
+            }}
+
+            public DataLoader() {{
+                this.conn = createConnection();
+            }}
+
+            public Stream<OutputRow> process(String query) {{
+                List<OutputRow> list = new ArrayList<>();
+                 try {{
+                    PreparedStatement stmt = this.conn.prepareStatement(query);
+                    stmt.setQueryTimeout({str(self.query_timeout)});
+                    stmt.setFetchSize({str(self.fetch_size)});
+                    try (ResultSet rs = stmt.executeQuery()) {{
+                        ResultSetMetaData meta = rs.getMetaData();
+                        int columnCount = meta.getColumnCount();
+                        while (rs.next()) {{
+                            OutputRow row = new OutputRow();
+                            {self.create_output_row_java_code()}
+                            list.add(row);
+                        }}
+                    }}
+                }} catch (SQLException e) {{
+                    throw new RuntimeException("SQL error: " + e.getMessage(), e);
+                }}
+
+                return list.stream();
+            }}
+
+            public Stream<OutputRow> endPartition() {{
+                return Stream.empty();
+            }}
+        }}
+        $$
+        ;
+        """
+
+        jdbc_udtf = f"""
+            select result.* from {partition_table}, table({jdbc_ingestion_name}({PARTITION_TABLE_COLUMN_NAME})) AS result
+            """
+
+        self.session.sql(jdbc_udtf_registration, _emit_ast=self._emit_ast).collect()
+        return BaseDriver.to_result_snowpark_df_udtf(
+            self.session.sql(jdbc_udtf, _emit_ast=self._emit_ast),
+            self.schema,
+            _emit_ast=self._emit_ast,
+        )
+
+    def generate_create_connection(self):
+        user_properties_overwrite = ""
+        if self.properties is not None:
+            user_properties_overwrite = "\n".join(
+                [
+                    f'properties.put("{key}", "{value}");'
+                    for key, value in self.properties.items()
+                ]
+            )
+        get_secret = """
+                    SnowflakeSecrets secrets = SnowflakeSecrets.newInstance();
+                    UsernamePassword up = secrets.getUsernamePassword("cred");
```

Comment:
does this throw error if "cred" is not available?

Reply:
yes

Reply:
In python code, secret name was fetched and assigned to 'cred' in java stored procedure, so this only raise an error when secret is not correctly setup in EAI